machines/backupAndRestore/backupAndRestoreSetup.ts

```diff
@@ -330,30 +344,24 @@ export const backupAndRestoreSetupMachine = model.createMachine(
     },
 
     services: {
-      checkInternet: async () => await NetInfo.fetch(),
       isUserSignedAlready: () => async () => {
         return await Cloud.isSignedInAlready();
       },
       signIn: () => async () => {
         return await Cloud.signIn();
       },
+      checkInternet: async () => await NetInfo.fetch(),
     },
 
     guards: {
-      isInternetConnected: (_, event) => event.data.isConnected,
+      isInternetConnected: (_, event) =>
+        !!(event.data as NetInfoState).isConnected,
       isNetworkError: (_, event) => event.data.error === NETWORK_REQUEST_FAILED,
       isSignedIn: (_context, event) =>
         (event.data as isSignedInResult).isSignedIn,
-      isNotSignedInIOS: (_context, event) => {
+      isIOSAndNotSignedIn: (_context, event) => {
         return !(event.data as isSignedInResult).isSignedIn && isIOS();
```

Comment:
Please invert the order of evaluation as on Android the isSignedInResult has a different structure iirc.